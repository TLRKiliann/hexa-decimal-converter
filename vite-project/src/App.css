#root {
  max-width: 1280px;
  margin: 0 auto;
  padding: 2rem;
  text-align: center;
}

.img--div {
  position: fixed;
  top: 0px;
  right: 0px;
  left: -5px;
  filter: opacity(80%);
  z-index: -1;
}

.img--div img {
  width: 100%;
  height: auto;
  object-fit: cover;
  opacity: 0.9;
}

.container {
  display: flex;
  justify-content: space-evenly;
  margin-top: 40px;
}

.container .box {
  width: 400px;
  display: flex;
  flex-direction: column;
  background-color: rgba(81,101,255, 50%);
  border-radius: 12px;
  box-shadow: 0px 0px 10px royalblue;
  padding: 10px 20px;
}

.container .box h2 {
  margin: auto;
  margin-top: 10px;
  margin-bottom: 20px;
  color: #00ffaa;
}

.container .box .labelinput--box {
  width: 320px;
  margin: auto;
  display: flex;
  align-items: center;
  justify-content: space-between;
  background-color: rgba(255,255,255, 10%);
  border-radius: 7px;
  padding-left: 10px;
}

.container .box .labelinput--box label {
  font-size: 1.0rem;
  font-weight: bold;
}

.container .box .labelinput--box input {
  width: 160px;
  font-size: 1.1rem;
  border: none;
  outline: none;
  border-radius: 5px;
  color: #3946aa;
  margin-left: 10px;
  padding: 2px 7px;
}

.container .box button {
  display: flex;
  justify-content: center;
  width: 140px;
  background: linear-gradient(45deg,#7887ff 1%, #3946aa 45% ,#7887ff 99%);
  font-weight: 600;
  color: #00ffaa;
  margin: 30px auto;
  box-shadow: 0px 0px 7px #90ee90;
}

.container .box .box--result {
  display: flex;
  background: #7887ff;
  border-radius: 7px;
  margin-bottom: 10px;
  padding-left: 20px;
}

.container .box .box--result p {
  font-size: 1.0rem;
  font-weight: bold;
}