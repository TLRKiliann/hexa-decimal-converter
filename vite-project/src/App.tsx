import CompBinaryDecimal from './components/CompBinaryDecimal.tsx'
import CompHexadecimalDecimal from './components/CompHexadecimalDecimal.tsx'
import CompAsciiToBinary from './components/CompAsciiToBinary.tsx'
import CompBinaryToAscii from './components/CompBinaryToAscii.tsx'
import CompAsciiToHex from './components/CompAsciiToHex.tsx'
import CompHexToAscii from './components/CompHexToAscii.tsx'
import math from '../public/images/math_wallpaper.jpeg'
import './App.css'

function App() {

  return (
    <>
      <div className="img--div">
        <img src={math} alt="math_wallpaper" />
      </div>
      <h1>Converter</h1>

      <div className="container">

        <div className="box">
          <h2>ASCII to binary:</h2>
          <CompAsciiToBinary />
        </div>

        <div className="box">
          <h2>binary to ASCII:</h2>
          <CompBinaryToAscii />
        </div>

      </div>

      <div className="container">

        <div className="box">
          <h2>binary to decimal:</h2>
          <CompBinaryDecimal />
        </div>

        <div className="box">
          <h2>hexadecimal to decimal:</h2>
          <CompHexadecimalDecimal />
        </div>

      </div>

      <div className="container">

        <div className="box">
          <h2>ASCII to Hex:</h2>
          <CompAsciiToHex />
        </div>

        <div className="box">
          <h2>Hex to ASCII:</h2>
          <CompHexToAscii />
        </div>

      </div>

    </>
  )
}

export default App
