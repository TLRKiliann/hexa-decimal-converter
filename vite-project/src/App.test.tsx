import { test, expect } from "vitest"
import { create } from "react-test-renderer"
import App from './App';

test('Snapshat of App', () => {
    const tree = create(<App />)
    expect(tree.toJSON()).toMatchSnapshot()
});