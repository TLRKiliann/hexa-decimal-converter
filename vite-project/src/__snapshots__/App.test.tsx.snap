// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Snapshat of App 1`] = `
[
  <div
    className="img--div"
  >
    <img
      alt="math_wallpaper"
      src="/public/images/math_wallpaper.jpeg"
    />
  </div>,
  <h1>
    Converter
  </h1>,
  <div
    className="container"
  >
    <div
      className="box"
    >
      <h2>
        ASCII to Binary:
      </h2>
      <div
        className="labelinput--box"
      >
        <label>
          ASCII
           :
        </label>
        <input
          data-testid="testid--input"
          onChange={[Function]}
          placeholder="ASCII"
          type="text"
          value=""
        />
      </div>
      <button
        data-testid="testid-btn"
        onClick={[Function]}
        type="button"
      >
        Convert
      </button>
      <div
        className="box--result"
      >
        <span
          data-testid="span-testid"
        >
          Result : 
          Not binary !!!
        </span>
      </div>
    </div>
    <div
      className="box"
    >
      <h2>
        Binary to ASCII:
      </h2>
      <div
        className="labelinput--box"
      >
        <label>
          Binary
           :
        </label>
        <input
          data-testid="testid--input"
          onChange={[Function]}
          placeholder="Binary"
          type="text"
          value=""
        />
      </div>
      <button
        data-testid="testid-btn"
        onClick={[Function]}
        type="button"
      >
        Convert
      </button>
      <div
        className="box--result"
      >
        <span
          data-testid="span-testid"
        >
          Result : 
          Not binary !!!
        </span>
      </div>
    </div>
  </div>,
  <div
    className="container"
  >
    <div
      className="box"
    >
      <h2>
        Binary to Decimal:
      </h2>
      <div
        className="labelinput--box"
      >
        <label>
          Binary
           :
        </label>
        <input
          data-testid="testid--input"
          onChange={[Function]}
          placeholder="Binary"
          type="text"
          value=""
        />
      </div>
      <button
        data-testid="testid-btn"
        onClick={[Function]}
        type="button"
      >
        Convert
      </button>
      <div
        className="box--result"
      >
        <span
          data-testid="span-testid"
        >
          Result : 
          Not binary !!!
        </span>
      </div>
    </div>
    <div
      className="box"
    >
      <h2>
        Hexadecimal to Decimal:
      </h2>
      <div
        className="labelinput--box"
      >
        <label>
          Hexadecimal
           :
        </label>
        <input
          data-testid="testid--input"
          onChange={[Function]}
          placeholder="Hexadecimal"
          type="text"
          value=""
        />
      </div>
      <button
        data-testid="testid-btn"
        onClick={[Function]}
        type="button"
      >
        Convert
      </button>
      <div
        className="box--result"
      >
        <span
          data-testid="span-testid"
        >
          Result : 
          Not binary !!!
        </span>
      </div>
    </div>
  </div>,
  <div
    className="container"
  >
    <div
      className="box"
    >
      <h2>
        ASCII to Hexadecimal:
      </h2>
      <div
        className="labelinput--box"
      >
        <label>
          ASCII
           :
        </label>
        <input
          data-testid="testid--input"
          onChange={[Function]}
          placeholder="ASCII"
          type="text"
          value=""
        />
      </div>
      <button
        data-testid="testid-btn"
        onClick={[Function]}
        type="button"
      >
        Convert
      </button>
      <div
        className="box--result"
      >
        <span
          data-testid="span-testid"
        >
          Result : 
          Not binary !!!
        </span>
      </div>
    </div>
    <div
      className="box"
    >
      <h2>
        Hexadecimal to ASCII:
      </h2>
      <div
        className="labelinput--box"
      >
        <label>
          Hexadecimal
           :
        </label>
        <input
          data-testid="testid--input"
          onChange={[Function]}
          placeholder="Hexadecimal"
          type="text"
          value=""
        />
      </div>
      <button
        data-testid="testid-btn"
        onClick={[Function]}
        type="button"
      >
        Convert
      </button>
      <div
        className="box--result"
      >
        <span
          data-testid="span-testid"
        >
          Result : 
          Not binary !!!
        </span>
      </div>
    </div>
  </div>,
]
`;
